File: comparing_languages/python/code.py
import sys
import random

def main():
    u = random.randint(1, 10)      
    r = random.randint(0, 10000) 
    a  = [0] * 10000             
    for i in range(10000):       
      for j in range(100000):    
        a[i] += j%u        
      a[i] += r           


if __name__ == "__main__":
    import time 
    start = time.time()
    main()
    duration = time.time() - start
    print(f"Python 用时: {duration}s")

File: comparing_languages/rust/code.rs
use rand::{thread_rng, Rng};
use std::time::Instant;

fn main() {
    let start = Instant::now();

    {   
        let n = thread_rng().gen_range(1..10);
        let r = thread_rng().gen_range(1..10000);

        let mut a = [0; 10000];
        for i in 0..10000 {
            for j in 0..100000 {
                a[i] += j % n
            }
            a[i] += r;
        }
    }    
    let duration = start.elapsed();
    println!("Rust 用时: {:?}", duration);
}